### PR TITLE
Queue timeout

### PR DESCRIPTION
**Pull request by casperisfine**
Add `merge.timeout` to the `shipit.yml`. Defaults to `1h`.

If a pull request can't be merged in that timeframe, it will be rejected.

Additionally, now that there is a `merge` namespace in `shipit.yml`, `ci.pr.require` and `ci.pr.ignore` have been renamed to `merge.require` and `merge.ignore` respectively.

@wvanbergen for review please.

### CODE REVIEW

File: app/models/shipit/deploy_spec.rb [hunk 1]
```diff
@@ -144,21 +144,27 @@ def soft_failing_statuses
...
-        Array.wrap(config('ci', 'pr', 'ignore'))
+      if config('merge', 'require') || config('merge', 'ignore')
+        Array.wrap(config('merge', 'ignore'))
       else
         soft_failing_statuses | hidden_statuses
       end
     end
 
+    def pull_request_timeout
+      Duration.parse(config('merge', 'require') { '1h' })
```
Comment (wvanbergen) [inline app/models/shipit/deploy_spec.rb]: `config('merge', 'timeout')`  ?

Comment (casperisfine) [inline app/models/shipit/deploy_spec.rb]: 🤦‍♂️

File: test/models/deploy_spec_test.rb [hunk 2]
```diff
@@ -398,10 +399,12 @@ class DeploySpecTest < ActiveSupport::TestCase
       assert_equal %w(ci/circleci ci/jenkins ci/travis).sort, @spec.pull_request_ignored_statuses.sort
     end
 
-    test "pull_request_ignored_statuses defaults to empty if `ci.pr` is present" do
+    test "pull_request_ignored_statuses defaults to empty if `ci.pr.require` is present" do
```
Comment (wvanbergen) [inline test/models/deploy_spec_test.rb]: `merge.require`

File: test/models/deploy_spec_test.rb [hunk 3]
```diff
@@ -411,8 +414,10 @@ class DeploySpecTest < ActiveSupport::TestCase
 
     test "pull_request_ignored_statuses returns `ci.pr.ignore` if present" do
```
Comment (wvanbergen) [inline test/models/deploy_spec_test.rb]: `merge.ignore`

File: app/models/shipit/pull_request.rb [hunk 4]
```diff
@@ -108,6 +112,7 @@ def reject!(reason)
     def reject_unless_mergeable!
       return reject!('merge_conflict') if merge_conflict?
       return reject!('ci_failing') unless all_status_checks_passed?
+      return reject!('timedout') if timedout?
```
Comment (wvanbergen) [inline app/models/shipit/pull_request.rb]: For my clarification: by having a minutely cron job, this will run not long after after the hour is over. Right?

Comment (casperisfine) [inline app/models/shipit/pull_request.rb]: Yes. Worst case 1h and 59 seconds. Unless there is an outage or something and crons or background queue is broken.

Anyway I'm about to submit another PR so that the job is triggered more often than once per minute.

**Comment by casperisfine** [comment]
@wvanbergen I fixed all the stupid mistakes.
